File: 088. Merge Sorted Array/088. Merge Sorted Array.cpp
//把数组依次从后往前放可以减少大量的数据移动。

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        nums1.resize(m+n);
        int p1 = m-1;
        int p2 = n-1;
        int tmp = m+n-1;
        while(p1+1&&p2+1){
            if(nums1[p1]>nums2[p2]){
                nums1[tmp--] = nums1[p1--];
            }else{
                nums1[tmp--] = nums2[p2--];
            }
        }
        if(p2+1)
            while(p2+1){
                nums1[p2] = nums2[p2];
                p2--;
            }
    }
};

File: 088. Merge Sorted Array/088. Merge Sorted Array.java
//从后向前考虑。如果A中的先遍历完，应把B中剩下的元素复制到A中。

public void merge(int[] nums1, int m, int[] nums2, int n) {
		int index = m + n - 1;
		int i = m - 1;
		int j = n - 1;
		while (i >= 0 && j >= 0)
			if (nums1[i] >= nums2[j]) {
				nums1[index] = nums1[i];
				i--;
				index--;
			} else {
				nums1[index] = nums2[j];
				j--;
				index--;
			}
		//继续遍历剩下的元素
		if (i == -1) {
			while (j >= 0) {
				nums1[index] = nums2[j];
				j--;
				index--;
			}
		}
	}

